### DISCUSSION TITLE
Crossplay

### DISCUSSION THREAD

**Discussion by m3ld0n** [lloesche/valheim-server-docker#506] [Q&A]
Anyone had any luck enabling crossplay support? The official guide mentiones using the "-crossplay" launch option. Not sure if that is a variable on the container or what do I need to do?

Edit1: discovered you can add the "-crossplay" command via the SERVER_ARGS variable. This makes the server change from using Steamworks to Playfab which is the new backend for crossplay, however it keeps disconnecting and reconnecting again. Tried running a server on my Windows machine with Steam and no issues there. I am thinking the container may be missing some libraries needed by this new backend. I suppose I will need to wait until the image is updated, unless someone has a workaround.

**Comment by fr3sh68** [comment]
Now that Valhiem is out on Xbox I have some friends that want to join my already existing dedicated server but am having this same issue.

**Reply by m3ld0n** [reply]
People on the official valheim discord are saying that the current crossplay implementation is unstable and a hotfix is on the way. Lets hope that fixes the issue once it arrives.

**Comment by SammyIsra** [comment]
Adding the `-crossplay` argument to `SERVER_ARGS` seems to make `valheim-server` hang on this step:
![image](https://user-images.githubusercontent.com/8962228/200356240-cefbcc90-7d68-4d96-a1d5-07f9dadcd1f3.png)

**Reply by m3ld0n** [reply]
Yep. I have those same logs.
Haven't been able to make cross-play work yet.

My server even appears on the Community tab in-game when crossplay is enabled, but no one can enter and the connection keeps looping.

![image](https://user-images.githubusercontent.com/3538883/200375156-b6841c32-5fce-415f-9b66-90867261a1fa.png)

**Comment by majcher666** [comment]
Any solution ?

**Comment by Christolito** [comment]
I'm getting kicked out of the server now and then, im guessing that it might be due to cross play, and the playfab / not the container/server? What do you think? See log: https://pastebin.com/By6GXrUW

**Reply by Christolito** [reply]
Looks like it's on the Valheim developer radar; https://valheimbugs.featureupvote.com/suggestions/334493/crossplay-multiplayer-issues#comment454163

**Comment by JWalk9000** [comment]
Any Updates on enabling crossplay?

**Comment by clarkent86** [comment]
I'm confused - isn't Valheim still unreleased on Xbox? I thought we were still waiting for Spring '23. I don't have an Xbox but want some of my friends to play.

**Reply by wcdanh2** [reply]
I think right now cross play is cross-store. People who bought it on steam can play with people who bought it through the MS store.

**Reply by clarkent86** [reply]
Ahh ok thanks for clarifying

**Comment by prime001** [comment]
I posted the solution mostly under issue 71 but also mentioned it in issue 70.
